pkg/kubelet/dockertools/manager.go

```diff
@@ -261,23 +271,53 @@ func (self *DockerManager) GetPodStatus(pod *api.Pod) (*api.PodStatus, error) {
 		}
 		dockerContainerName := dockerName.ContainerName
 		c, found := expectedContainers[dockerContainerName]
-		terminationMessagePath := ""
 		if !found {
-			// TODO(dchen1107): should figure out why not continue here
-			// continue
-		} else {
-			terminationMessagePath = c.TerminationMessagePath
+			continue
 		}
-		// We assume docker return us a list of containers in time order
+		terminationMessagePath := c.TerminationMessagePath
+		if containerDone.Has(dockerContainerName) {
+			continue
+		}
+		// We assume docker returns a list of containers sorted reversedly by
+		// time.
 		if containerStatus, found := statuses[dockerContainerName]; found {
```

Comment:
Changed to use a pointer. I wholeheartedly agree that the flow here can be improved. It's very easy to break something. We should clean up the code sometime.